.github/workflows/e2e.yaml

```diff
@@ -0,0 +1,115 @@
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          gh api \
+            --method POST \
+            -H "Accept: application/vnd.github+json" \
+            /repos/${{ github.repository }}/statuses/${{ steps.checkout-pr.outputs.commit-sha }} \
+            -f state='pending' \
+            -f target_url='https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}' \
+            -f description='Running...' \
+            -f context="E2E"
+
+  e2e:
+    runs-on: ubuntu-22.04
+    needs: check-triggered
+    if: ${{ github.event.issue.pull_request && contains(github.event.comment.body, '/test-e2e') }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          submodules: true
+
+      - name: Checkout PR
+        id: checkout-pr
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          gh pr checkout ${{ github.event.issue.number }}
+          git submodule update --init --recursive
+          OUTPUT=$(git log -1 --format='%H')
+          echo "::set-output name=commit-sha::$OUTPUT"
+
+      - name: Fetch History
+        run: git fetch --prune --unshallow
+
+      - name: End to end
+        run: make e2e
+        env:
+          UPTEST_AZURE_CREDS: ${{ secrets.UPTEST_AZURE_CREDS }}
+
+      - name: Create Successful Status Check
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          gh api \
+            --method POST \
+            -H "Accept: application/vnd.github+json" \
+            /repos/${{ github.repository }}/statuses/${{ steps.checkout-pr.outputs.commit-sha }} \
+            -f state='success' \
+            -f target_url='https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}' \
+            -f description='Passed' \
+            -f context="E2E"
+
+      - name: Collect Cluster Dump
+        if: always()
+        run: |
+          export CONTROLPLANE_DUMP_DIRECTORY=./_output/controlplane-dump
+          make controlplane.dump
+
+      - name: Upload Cluster Dump
+        if: always()
+        uses: actions/upload-artifact@v3
+        with:
+          name: cluster-dump
+          path: ./_output/controlplane-dump
+
+      - name: Cleanup
+        if: always()
+        run: |
+          eval $(make --no-print-directory build.vars)
+          ${KUBECTL} delete managed --all
+
+      - name: Create Unsuccessful Status Check
+        if: failure()
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          gh api \
+            --method POST \
+            -H "Accept: application/vnd.github+json" \
+            /repos/${{ github.repository }}/statuses/${{ steps.checkout-pr.outputs.commit-sha }} \
+            -f state='failure' \
+            -f target_url='https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}' \
+            -f description='Failed' \
+            -f context="E2E"
```

Comment:
It seems symmetric between platform refs. Can we do [reusable workflow](https://docs.github.com/en/actions/using-workflows/reusing-workflows) out of e2e.yaml , store it centrally and reuse it across all tested Configuration repo?

Reply:
Thanks, I will check that.
FWIW, I mostly copied them providers so investing for reusable workflows would improve the codebase there as well.

Reply:
@turkenh another thought, wrapping the whole thing into [composite action](https://docs.github.com/en/actions/creating-actions/creating-a-composite-action) might be even better from a reusability standpoint. Anybody from the community will be able to include e.g. `uptest-e2e` action and following file path conventions it will trigger the whole e2e flow without much additional configuration

Reply:
@ytsarev I like that but wondering if we should handle that with a separate ticket instead of blocking e2e testing work? Would you mind creating a ticket for that in https://github.com/upbound/uptest?

Reply:
@turkenh agreed, it shouldn't be blocking , https://github.com/upbound/uptest/issues/31 created